routes/originConda.js

```diff
@@ -26,13 +26,13 @@ async function fetchCondaChannelData(channel) {
 
 async function fetchCondaRepoData(channel, subdir) {
   const key = `${channel}-${subdir}-repoData`
-  let channelData = condaCache.get(key)
-  if (!channelData) {
+  let repoData = condaCache.get(key)
+  if (!repoData) {
     const url = `${condaChannels[channel]}/${subdir}/repodata.json`
-    channelData = await requestPromise({ url, method: 'GET', json: true })
-    condaCache.put(key, channelData, 8 * 60 * 60 * 1000) // 8 hours
+    repoData = await requestPromise({ url, method: 'GET', json: true })
```

Comment:
Tested origins/conda/conda-forge/_go_select/revisions  timed out 30 sec.  May want to consider downloading the compressed files, e.g. repodata.json.bz2, instead.

Reply:
In addition, /origins series of end points typically reflect the namespace in the coordinates, see originNpm, originMaven and originGitHub.  If subdir is used for namespace, to be consistent, it should be reflected in both end points in origins/conda, i.e. origins/conda/channel/subdir/name and  origins/conda/channel/subdir/name/revisions. This will help performance here as well.

Reply:
> Tested origins/conda/conda-forge/_go_select/revisions timed out 30 sec. Does downloading the compressed files, e.g. repodata.json.bz2, instead of repodata.json help?

timed out? that's probably for the first request, it needs to fetch and parse about 80MB of json, which I don't think is a network bottleneck. it should only happen after the first download though.

Reply:
> In addition, /origins series of end points typically reflect the namespace in the coordinates, see originNpm, originMaven and originGitHub. If subdir is used for namespace, to be consistent, it should be reflected in both end points in origins/conda, i.e. origins/conda/channel/subdir/name and origins/conda/channel/subdir/name/revisions. This will help performance here as well.

From a usability POV, it's common to want to know what architecture and what versions are available, as I said earlier, its not exactly a namespace but an architecture.
a more suitable approach would be to make the namespace optional in which all the namespaces are checked if none is provided, or only the specified ones.

Reply:
You are right from a usability point of view.  Adding an optional namespace sounds good.  In addition, downloading the compressed .bz2 file speeds up the performance as well: 1.7 sec for https://conda.anaconda.org/conda-forge/linux-64/repodata.json.bz2 (30MB) vs. ~7sec for non compressed repo.json (230MB).